blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/RetryNTimes.java

```diff
@@ -0,0 +1,93 @@
+package com.rackspacecloud.blueflood.io.datastax;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.exceptions.DriverException;
+import com.datastax.driver.core.policies.RetryPolicy;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.WriteType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A class that retry on read/write/unavailable timeouts n times, where
+ * n is a configurable number
+ */
+public class RetryNTimes implements RetryPolicy {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RetryNTimes.class);
+    private final int readAttempts;
+    private final int writeAttempts;
+    private final int unavailableAttempts;
+
+    /**
+     * A Datastax {@link RetryPolicy} that is capable of retrying n times
+     * @param readAttempts
+     * @param writeAttempts
+     * @param unavailableAttempts
+     */
+    public RetryNTimes(int readAttempts, int writeAttempts, int unavailableAttempts) {
+        this.readAttempts = readAttempts;
+        this.writeAttempts = writeAttempts;
+        this.unavailableAttempts = unavailableAttempts;
+    }
+
+    @Override
+    public RetryDecision onReadTimeout(Statement stmnt, ConsistencyLevel cl,
+                                       int requiredResponses, int receivedResponses,
+                                       boolean dataReceived, int rTime) {
+        if (dataReceived) {
+            return RetryDecision.ignore();
+        } else if (rTime < readAttempts) {
```

Comment:
It will retry 3 times, when rTime is:
* 0 < 3
* 1 < 3
* 2 < 3

When rTime reaches 3, it will go to the else and do a rethrow()